contracts/vesting/src/tests/other_operations.rs

```diff
@@ -0,0 +1,1006 @@
+    vesting_client.initialize(
+        &admin,
+        &vesting_token,
+        &vesting_balances,
+        &None,
+        &None,
+        &10u32,
+    );
+
+    assert_eq!(vesting_client.query_vesting_total_supply(), 1_000);
+
+    vesting_client.burn(&vester1, &500);
+
+    assert_eq!(vesting_client.query_vesting_total_supply(), 500);
+    assert_eq!(token.balance(&vester1), 500);
+}
+
+#[test]
+#[should_panic(expected = "Vesting: Burn: Invalid burn amount")]
+fn burn_should_panic_when_invalid_amount() {
+    let env = Env::default();
+    env.mock_all_auths();
+    env.budget().reset_unlimited();
+
+    let admin = Address::generate(&env);
+    let vester1 = Address::generate(&env);
+    let token = deploy_token_contract(&env, &admin);
+
+    token.mint(&vester1, &1_000);
+
+    let vesting_token = VestingTokenInfo {
+        name: String::from_str(&env, "Phoenix"),
+        symbol: String::from_str(&env, "PHO"),
+        decimals: 6,
+        address: token.address.clone(),
+        total_supply: 0,
+    };
+    let vesting_balances = vec![
+        &env,
+        VestingBalance {
+            address: vester1.clone(),
+            balance: 1_000,
+            curve: Curve::SaturatingLinear(SaturatingLinear {
+                min_x: 15,
+                min_y: 120,
+                max_x: 60,
+                max_y: 0,
+            }),
+        },
+    ];
+
+    let vesting_client = instantiate_vesting_client(&env);
+    vesting_client.initialize(
+        &admin,
+        &vesting_token,
+        &vesting_balances,
+        &None,
+        &None,
+        &10u32,
+    );
+
+    vesting_client.burn(&vester1, &0);
+}
+
+#[test]
+#[should_panic(expected = "Vesting: Burn: Critical error - total supply cannot be negative")]
+fn burn_should_panic_when_total_supply_becomes_negative() {
+    const TOO_MUCH: i128 = 1_000;
+    let env = Env::default();
+    env.mock_all_auths();
+    env.budget().reset_unlimited();
+
+    let admin = Address::generate(&env);
+    let vester1 = Address::generate(&env);
+    let token = deploy_token_contract(&env, &admin);
+
+    token.mint(&vester1, &1_300);
+
+    let vesting_token = VestingTokenInfo {
+        name: String::from_str(&env, "Phoenix"),
+        symbol: String::from_str(&env, "PHO"),
+        decimals: 6,
+        address: token.address.clone(),
+        total_supply: 0,
+    };
+    let vesting_balances = vec![
+        &env,
+        VestingBalance {
+            address: vester1.clone(),
+            balance: 500,
+            curve: Curve::SaturatingLinear(SaturatingLinear {
+                min_x: 15,
+                min_y: 120,
+                max_x: 60,
+                max_y: 0,
+            }),
+        },
+    ];
+
+    let vesting_client = instantiate_vesting_client(&env);
+    vesting_client.initialize(
+        &admin,
+        &vesting_token,
+        &vesting_balances,
+        &None,
+        &None,
+        &10u32,
+    );
+
+    vesting_client.burn(&vester1, &TOO_MUCH);
```

Comment:
```suggestion
            balance: 500,
            curve: Curve::SaturatingLinear(SaturatingLinear {
                min_x: 15,
                min_y: 120,
                max_x: 60,
                max_y: 0,
            }),
        },
    ];

    let vesting_client = instantiate_vesting_client(&env);
    vesting_client.initialize(
        &admin,
        &vesting_token,
        &vesting_balances,
        &None,
        &None,
        &10u32,
    );

    vesting_client.burn(&vester1, &600);
```
Just assign it, here it works better.